Is OkHttp no longer optional?

Upgrading to the 2.6.0 version now gives he following error : 

java.lang.NoClassDefFoundError: com.squareup.okhttp.Authenticator
	at com.cloudant.client.org.lightcouch.CouchDbClient.<init>(CouchDbClient.java:95)
	at com.cloudant.client.org.lightcouch.CouchDbClient.<init>(CouchDbClient.java:140)
	at com.cloudant.client.api.CloudantClient.<init>(CloudantClient.java:138)
	at com.cloudant.client.api.ClientBuilder.build(ClientBuilder.java:330)
...
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:95)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:507)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
Caused by: java.lang.ClassNotFoundException: com.squareup.okhttp.Authenticator
	at java.net.URLClassLoader.findClass(URLClassLoader.java:607)
	at java.lang.ClassLoader.loadClassHelper(ClassLoader.java:797)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:771)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:325)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:752)
	... 23 more



- CloudantClient (java-cloudant) version(s) that are affected by this issue. : 2.6.0
- Java version (including vendor and platform). IBM JDK 1.8
- If you're using the optional okhttp dependency. No

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,3 +1,7 @@
+# Unreleased
+- [FIX] `NoClassDefFoundError: com.squareup.okhttp.Authenticator` for version 2.6.0 if the optional
+  okhttp dependency was not included.
+
 # 2.6.0 (2016-09-12)
 - [NEW] Enabled `reduce` and other reduce related parameters to be set when using
   `MultipleRequestBuilder`.

--- a/cloudant-client/src/main/java/com/cloudant/client/api/ClientBuilder.java
+++ b/cloudant-client/src/main/java/com/cloudant/client/api/ClientBuilder.java
@@ -1,3 +1,17 @@
+/*
+ * Copyright © 2015, 2016 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
 package com.cloudant.client.api;
 
 import com.cloudant.client.api.views.Key;

--- a/cloudant-http/src/main/java/com/cloudant/http/internal/ok/OkHttpClientHttpUrlConnectionFactory.java
+++ b/cloudant-http/src/main/java/com/cloudant/http/internal/ok/OkHttpClientHttpUrlConnectionFactory.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2015 IBM Corp. All rights reserved.
+ * Copyright © 2015, 2016 IBM Corp. All rights reserved.
  *
  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
  * except in compliance with the License. You may obtain a copy of the License at
@@ -14,20 +14,15 @@
 
 package com.cloudant.http.internal.ok;
 
-import com.cloudant.http.interceptors.ProxyAuthInterceptor;
 import com.cloudant.http.internal.DefaultHttpUrlConnectionFactory;
-import com.squareup.okhttp.Authenticator;
 import com.squareup.okhttp.ConnectionSpec;
 import com.squareup.okhttp.Credentials;
 import com.squareup.okhttp.OkHttpClient;
 import com.squareup.okhttp.OkUrlFactory;
-import com.squareup.okhttp.Request;
-import com.squareup.okhttp.Response;
 
 import java.io.IOException;
 import java.net.HttpURLConnection;
 import java.net.PasswordAuthentication;
-import java.net.Proxy;
 import java.net.URL;
 import java.util.Arrays;
 import java.util.logging.Logger;
@@ -93,32 +88,6 @@ public class OkHttpClientHttpUrlConnectionFactory extends DefaultHttpUrlConnecti
 
     public OkHttpClient getOkHttpClient() {
         return client;
-    }
-
-    private static class ProxyAuthenticator implements Authenticator {
-
-        private final String creds;
-
-        ProxyAuthenticator(String creds) {
-            this.creds = creds;
-        }
-
-        @Override
-        public Request authenticate(Proxy proxy, Response response) throws IOException {
-            // Don't interfere with normal auth, this is just for proxies.
-            return null;
-        }
-
-        @Override
-        public Request authenticateProxy(Proxy proxy, Response response) throws IOException {
-            if (creds.equals(response.request().header("Proxy-Authorization"))) {
-                // If the proxy creds have already been tried then give up
-                return null;
-            } else {
-                return response.request().newBuilder().addHeader(ProxyAuthInterceptor
-                        .PROXY_AUTH_HEADER, creds).build();
-            }
-        }
     }
 
 }

--- a/cloudant-http/src/main/java/com/cloudant/http/internal/ok/ProxyAuthenticator.java
+++ b/cloudant-http/src/main/java/com/cloudant/http/internal/ok/ProxyAuthenticator.java
@@ -1,0 +1,49 @@
+/*
+ * Copyright © 2016 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.http.internal.ok;
+
+import com.cloudant.http.interceptors.ProxyAuthInterceptor;
+import com.squareup.okhttp.Authenticator;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+
+import java.io.IOException;
+import java.net.Proxy;
+
+class ProxyAuthenticator implements Authenticator {
+
+    private final String creds;
+
+    ProxyAuthenticator(String creds) {
+        this.creds = creds;
+    }
+
+    @Override
+    public Request authenticate(Proxy proxy, Response response) throws IOException {
+        // Don't interfere with normal auth, this is just for proxies.
+        return null;
+    }
+
+    @Override
+    public Request authenticateProxy(Proxy proxy, Response response) throws IOException {
+        if (creds.equals(response.request().header("Proxy-Authorization"))) {
+            // If the proxy creds have already been tried then give up
+            return null;
+        } else {
+            return response.request().newBuilder().addHeader(ProxyAuthInterceptor
+                    .PROXY_AUTH_HEADER, creds).build();
+        }
+    }
+}

